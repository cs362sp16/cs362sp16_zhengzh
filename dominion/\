#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "rngs.h"
#include "dominion.h"

int main(int argc, char *argv[]){

  PutSeed((long)argv[1]);

  struct gameState g;

  int k[10] = {embargo,adventurer,gardens,smithy,cutpurse,mine,ambassador,
	       outpost,baron,tribute};

  int numPlayers;
  int seed;
  int i;

  srand(time(NULL));

  for(i=0; i < 1000; i++){
	
	numPlayers = rand() % 4;
	seed = rand();

	initializeGame(numPlayers, k, seed, &g);

	printf("Begin Test ");
	printf("%d", i);
	printf(": \n");
	

		g.deckCount[0]=rand() % MAX_DECK;
		g.discardCount[0]=rand() % MAX_DECK;
		g.handCount[0]=rand() % MAX_HAND;

		int deckcount = g.deckCount[0];
		int discardcount = g.discardCount[0];
		int handcount = g.handCount[0];

		g.hand[0][0] = adventurer;

		int returnval = playCard(0,0,0,0,&g);
	
		if(returnval != 0){
		
			printf("Card did not execute, test failed\n");

		}

		if(g.handCount[0] <= handcount){

			printf("Was not able to draw card, test failed\n");
		
		}
		
	


		

  }

	return 0;
}
