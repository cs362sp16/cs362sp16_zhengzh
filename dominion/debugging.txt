The process of identifying and fixing bugs in my own code starts with a bug report from Chen Chongxian. This bug report is called zhengzh_bug2.txt. It states that my smithy function does not properly draw 3 cards, instead it draws 25 cards. On top of that, when it is called by cardEffects, it does not properly return 0. Following his bug report, I rewrote my cardtest1.c test file to see if it is true. It turns out that it does indeed draw 25 cards and does not return a 0 when called from cardEffect. I went back to my dominion.c code and modified a for loop and then added a return 0 after all my function calls in cardEffect. I ran my debugger again and it still failed the test for drawing cards. Then I realized that my debugger is wrong. This is because it was testing to see if the hand size if +3 bigger than what it originally was but it should be checking to see if it is +2 bigger since the card that is played is now discarded. After I changed that it seemed to fix it since my debugger now says that cardEffect returns 0 and that the draw test is also successful. The Agan's principles I used are "Getting a fresh view" since having Chongxian look at my code helped me identify that bug. Another principle is "Quit thinking and look", this applied since I did not think about why my function was not returning 0 at first. I was fairly sure of why it was not drawing 3 cards but not did even consider that it was not returning 0 when being called by cardEffect. Another Agan's principle that applies in this case is "Check the plugs". This applies since my debugger was actually wrong. Most importantly the Agan's rule of "Understand the system" applies. This is because I did not understand that cardEffect must return 0 and when I refactored I did not take that into consideration.

Below is the result of my successful test:


Test passed for drawing 3 cards
cardEffect successfully returned 0
Smithy successfully ran
