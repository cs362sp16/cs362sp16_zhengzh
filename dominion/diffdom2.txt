Starting game ...
Number of players is: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 5
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 6
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 0
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 0
The turn of player: 3
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 0
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 0
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 6
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 6
The turn of player: 3
You have this much money: 0
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 6
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 6
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 6
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
Playing a card
The turn of player: 2
You have this much money: 1
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 5
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 2
You have this much money: 5
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 0
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 9
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 1
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 3
The turn of player: 2
You have this much money: 3
The turn of player: 3
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 2
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
Playing a card
The turn of player: 2
You have this much money: 2
The turn of player: 3
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 5
The turn of player: 2
You have this much money: 6
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 1
Playing a card
The turn of player: 2
You have this much money: 4
The turn of player: 3
You have this much money: 2
The turn of player: 0
You have this much money: 6
Score for player 0: 1
Score for player 1: -4
Score for player 2: 3
Score for player 3: -4
