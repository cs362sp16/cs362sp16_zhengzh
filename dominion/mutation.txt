I did mutation testing by generating the mutants myself and then performing my unit and card tests on them. 

Here are the results:



Does not correctly draw 3 cards
cardEffect successfully returned 0
Great Hall is implemented incorrectly, however somewhere else in the code is making it work
error: no extra turn for user
Error incrementing coin
Error adding curse
No bugs in this supplyCount //

hand before
1
hand before
4
hand before
4
hand before
4
hand after
1
hand after
4
hand after
4
hand after
4
hand after
21
Can properly increment deckCount and handCount, also tested that it can add the correct card to its hand. However, for some reason the initialize function starts the game with player 0 instead of player 1 which can get confusing. This is an issue with initialize howevergainCard successfully ran//
0Test can end a player's turn, and loops back around when it reaches the max player, however, player number starts at player 0, ends at player 1, instead of player 1 and player 2.endTurn successfully ran//

All the test results with // at the end of them means that the test declares that the mutant is correct. This means that the test is inaccurate. So this occured in 3 out of 8 of my tests. The reason that these tests are not very good is because they only test 1 or 2 criterias and passes the function if those criterias are met. However, it does not check all the criterias for a passing function. Therefore some of these tests are inadequete and was quickly called out with mutation testing. 
