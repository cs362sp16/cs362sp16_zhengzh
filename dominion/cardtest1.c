/*Unit Test for the card Smithy
 *
 *
 */
#include "dominion.h"
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

int error =0;
void checkasserts(){
	if(error==0){
	printf("Smithy successfully ran\n");
	}
}

int main(){

  struct gameState g;

  int k[10] = {smithy,adventurer,gardens,embargo,cutpurse,mine,ambassador,
	       outpost,baron,tribute};

  int r = initializeGame(2, k, 5, &g);

  g.hand[0][0]=smithy;
  int numCardHandBefore=g.handCount[0];
  playCard(0,0,0,0, &g);
  if(g.handCount[0]!=numCardHandBefore+3){
	printf("Does not correctly draw 3 cards\n");
	error=1;	


	}//Does not correctly draw 3 cards
 
  checkasserts();
  return 0; 
}
