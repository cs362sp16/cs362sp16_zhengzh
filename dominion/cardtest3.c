
#include "dominion.h"
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

int error =0;
void checkasserts(){
	if(error==0){
	printf("isGameOver successfully ran\n");
	}
}

int main(){

  struct gameState g;

  int k[10] = {smithy,adventurer,gardens,embargo,cutpurse,mine,ambassador,
	       outpost,baron,tribute};

  int r = initializeGame(2, k, 5, &g);

  
  g.hand[0][0]=outpost;

  playCard(0,0,0,0, &g);

  if(g.outpostPlayed!=1){
	printf("Outpost does not increment correctly\n");
	error=1;
	}

  endTurn(&g);
  if(g.whoseTurn != 0){
	printf("error: no extra turn for user\n");
	error=1;
	}//This assertion fails, does not let user take an extra turn

  checkasserts();
return 0;

}
