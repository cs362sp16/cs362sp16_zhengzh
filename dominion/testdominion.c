#include <stdlib.h>
#include <stdio.h>
#include "dominion.h"

void randomize(int *array, size_t n);
void chooseCard(int k[10]);

void randomize(int *array, size_t n)
{
    if (n > 1) 
    {
        size_t i;
        for (i = 0; i < n - 1; i++) 
        {
          size_t j = i + rand() / (RAND_MAX / (n - i) + 1);
          int t = array[j];
          array[j] = array[i];
          array[i] = t;
        }
    }
}

void chooseCard(int k[10]){
	
	int i;
	int allCards[]={adventurer, council_room, feast, gardens, mine, remodel, smithy, village,baron, great_hall, minion, steward, tribute, ambassador, cutpurse, embargo, outpost, salvager, sea_hag, treasure_map};

	randomize(allCards, 20);

	for (i=0; i<10; i++){

		k[i]=allCards[i];

	}	
	

}

int main(int argc, char *argv[]){

	int seed = atoi(argv[1]);
	srand(seed);
	int winner[4];
	int numPlayer = (rand() % (4-2+1))+2;
	struct gameState g;
	int k[10];

	chooseCard(k);

	printf("Starting game ...\n");
	initializeGame(numPlayer, k, seed, &g);

        int money = 0;
        int i=0;
        int numSmithies = 0;
        int numAdventurers = 0;
	int player;

	printf("Number of players is: ");
	printf("%d", numPlayer);
	printf("\n");

	while(!isGameOver(&g)){

		player= whoseTurn(&g);
		printf("The turn of player: ");
		printf("%d", player);	
		printf("\n");	

		money = 0;
		
		
    	for (i = 0; i < numHandCards(&g); i++) {
      	if (handCard(i, &g) == copper)
    	money++;
      	else if (handCard(i, &g) == silver)
    	money += 2;
      	else if (handCard(i, &g) == gold)
    	money += 3;

	}

	printf("You have this much money: ");
	printf("%d", money);
	printf("\n");

	while (g.numActions > 0 ){
		
		for(i=0; i<numHandCards(&g); i++){

			if(handCard(i, &g)< 27 && handCard(i, &g) > 6){
				printf("Playing a card\n");
				playCard(i, 0, 0, 0, &g);
				break;	

			}


		}		

		g.numActions = g.numActions-1;	

	}
	
	int provinceorkingdom = rand() % 2;

	if(provinceorkingdom > 1){
	if(money >= 8){

		printf("Bought Province\n");
		buyCard(province, &g);
	}
	else if(money >= 6){

		printf("Bought Gold\n");
		buyCard(gold, &g);
	}
	else if(money >=3){

		printf("Bought Silver\n");
		buyCard(silver, &g);

	}
	}

	if(provinceorkingdom <=1){

		int buyNum =  rand() % 10;
		("Buying a Card\n");
		buyCard(buyNum, &g);

	}

	endTurn(&g);
	}

	for(i=0; i<numPlayer; i++){
	int score= scoreFor(i, &g);
	printf("Score for player ");
	printf("%d", i);
	printf(": ");
	printf("%d", score);
	printf("\n");
	}

return 0;

}
