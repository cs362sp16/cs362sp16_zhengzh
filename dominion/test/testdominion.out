Starting game ...
Number of players is: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
Playing a card
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 7
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 1
The turn of player: 0
You have this much money: 7
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
Playing a card
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 5
Playing a card
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 7
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
Playing a card
The turn of player: 1
You have this much money: 4
Playing a card
The turn of player: 0
You have this much money: 1
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 7
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 7
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 7
The turn of player: 0
You have this much money: 7
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 0
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 8
The turn of player: 0
You have this much money: 6
The turn of player: 1
You have this much money: 9
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
The turn of player: 0
You have this much money: 7
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 3
Playing a card
The turn of player: 0
You have this much money: 2
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 4
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 1
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 2
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 5
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 5
The turn of player: 1
You have this much money: 4
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
The turn of player: 0
You have this much money: 3
The turn of player: 1
You have this much money: 6
Score for player 0: 22
Score for player 1: 1
File 'dominion.c'
No executable lines
Removing 'dominion.c.gcov'

